cocos/animation/core/transform-array.ts

```diff
@@ -0,0 +1,227 @@
+        const baseOffset = TRANSFORM_STRIDE_IN_FLOATS * index;
+        Vec3.toArray(data, value, baseOffset + SCALE_OFFSET);
+    }
+
+    /**
+     * Same algorithm as https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/copyWithin
+     * except for the the operating objects are transforms.
+     */
+    public copyWithin (target: number, start: number, end?: number) {
+        this._data.copyWithin(
+            target * TRANSFORM_STRIDE_IN_FLOATS,
+            start * TRANSFORM_STRIDE_IN_FLOATS,
+            typeof end === 'number' ? end * TRANSFORM_STRIDE_IN_FLOATS : undefined,
+        );
+    }
+
+    /**
+     * Same algorithm as https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/fill
+     * except for the the operating objects are transforms.
+     */
+    public fill (value: Readonly<Transform>, start?: number, end?: number) {
+        const { length } = this;
+        start ??= 0;
+        end ??= length;
+        if (start >= length) {
+            return;
+        }
+        this.setTransform(start, value);
+        for (let i = start + 1; i < end; ++i) {
+            this.copyWithin(i, start, start + 1);
+        }
+    }
+
+    /**
+     * Same as `this.fill(Transform.ZERO, start, end)`.
+     */
+    public fillZero (start?: number, end?: number) {
+        this._data.fill(
+            0.0,
+            typeof start === 'number' ? start * TRANSFORM_STRIDE_IN_FLOATS : undefined,
+            typeof end === 'number' ? end * TRANSFORM_STRIDE_IN_FLOATS : undefined,
+        );
+    }
+
+    /**
+     * Same algorithm as https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/set
+     * except for:
+     * - the the operating objects are transforms,
+     * - plain array is not allowed.
+     */
+    public set (transformArray: Readonly<TransformArray>, targetOffset?: number) {
+        this._data.set(
+            (transformArray as TransformArray)._data,
+            typeof targetOffset === 'number' ? targetOffset * TRANSFORM_STRIDE_IN_FLOATS : undefined,
+        );
+    }
+
+    /**
+     * Same algorithm as https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice
+     * except for the the operating objects are transforms.
+     */
+    public slice (start?: number, end?: number) {
+        const dataSliced = this._data.slice(
+            typeof start === 'number' ? start * TRANSFORM_STRIDE_IN_FLOATS : undefined,
+            typeof end === 'number' ? end * TRANSFORM_STRIDE_IN_FLOATS : undefined,
+        );
+        return new TransformArray(dataSliced.buffer, dataSliced.byteOffset, dataSliced.length / TRANSFORM_STRIDE_IN_FLOATS);
+    }
+
+    /**
+     * Copy a span of `source` into `this`.
+     * Equivalent to `this.set(source.slice(sourceOffset, sourceOffset + size), targetOffset)`
+     * except without perform the slicing.
+     * */
+    public setSpan (
```

Comment:
The name is not easy to understand，I think it's functionality is `copyInto` or `fastCopyInto` ?

Reply:
Within the name, `set` is referenced from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/set . `span` is a synonym of `slice`, taken from https://en.cppreference.com/w/cpp/container/span . It wants to express: copy a subrange of dest array into this array at specified offset.

How do you feel about:

- `setSlice` (according to document)
- `copyRange`

Reply:
`setRange` and `copyRange` are better for me.

Reply:
https://github.com/cocos/cocos-engine/pull/13503/commits/38d60e6334156bccb6431b4be67d004266fa9bda